src/components/AttachmentPicker/index.native.js

```diff
@@ -177,30 +178,36 @@ class AttachmentPicker extends Component {
       * Common image picker handling
       *
       * @param {function} imagePickerFunc - RNImagePicker.launchCamera or RNImagePicker.launchImageLibrary
+      * @param {Boolean} isCamera
       * @returns {Promise<ImagePickerResponse>}
       */
-    showImagePicker(imagePickerFunc) {
+    showImagePicker(imagePickerFunc, isCamera = false) {
         return new Promise((resolve, reject) => {
-            imagePickerFunc(getImagePickerOptions(this.props.type), (response) => {
-                if (response.didCancel) {
-                    // When the user cancelled resolve with no attachment
-                    return resolve();
-                }
-                if (response.errorCode) {
-                    switch (response.errorCode) {
-                        case 'permission':
-                            this.showPermissionsAlert();
-                            break;
-                        default:
-                            this.showGeneralAlert();
-                            break;
-                    }
-
-                    return reject(new Error(`Error during attachment selection: ${response.errorMessage}`));
-                }
+            if (!isCamera) {
+                this.openImagePicker(imagePickerFunc, resolve, reject);
```

Comment:
I am generally against passing functions as params, especially when they have side-effects. for example in the `else` block no resolve can lead to a wrap when we would want to refactor `this.showPermissionAlert`.

Can get a Promise chain and let `openImagePicker` be an async function or have its own Promise to resolve.

Reply:
I feel it looks okay but let's take some comment from @Julesssss 
CC @mananjadhav

Reply:
> no resolve can lead to a wrap

@mananjadhav would you mind explaining this part? 

I agree it's not ideal currently. I would be interested too see this change too: 

> Can get a Promise chain and let openImagePicker be an async function or have its own Promise to resolve.

Reply:
> @mananjadhav would you mind explaining this part?

All I meant here is if we're defining a Promise, lets have either return a Promise chain or call with `resolve`/`reject`